src/main/java/uk/ac/ebi/ena/readtools/loader/common/InvalidBaseCharacterException.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2010-2020 EMBL - European Bioinformatics Institute
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package uk.ac.ebi.ena.readtools.loader.common;
+
+import java.util.Collection;
+
+public class InvalidBaseCharacterException extends RuntimeException {
+    private final String bases;
+
+    private final Collection<Character> invalidCharacters;
+
+    public InvalidBaseCharacterException(String bases, Collection<Character> invalidCharacters) {
+        this.bases = bases;
+        this.invalidCharacters = invalidCharacters;
+    }
+
+    public String getBases() {
+        return bases;
+    }
+
+    public Collection<Character> getInvalidCharacters() {
+        return invalidCharacters;
+    }
+
+    @Override
+    public String getMessage() {
+        return String.format("Invalid characters %s in bases : %s", invalidCharacters, bases);

```

Comment:
Ok, this is nice. Just to note that we might have tens of thousands or more bases coming from Oxford Nanopore of PacBio and not sure if it is viable to show the bases in the error message. It might be better to show the read name instead.

Reply:
the bases meant to go here would only belong to a particular read and as this is a kind of error thats supposed to stop whatever is going on, i dont think huge amount of data coming is going to be a problem. i also think its helpful to log the actual base next to invalid characters.